### PR TITLE
Adding prettier to project. Running prettier in all project

### LINKED ISSUE #444: Add Prettier
Reference implementation: https://github.com/styled-components/styled-components/pull/593

### PR DESCRIPTION
**Pull request by onluiz**
# Description

fix #444 

- Prettier added to project
- All files were formatted
- Each file changed will be formatted by prettier before each commit
- Rule of semi-column was kept

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- Tested running npm test
- Tested running npm start in docs directory

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have not generated a new package version. (the maintainers will handle that)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -66,7 +67,16 @@
   },
   "dependencies": {
     "classnames": "^2.2.5",
+    "husky": "^0.14.3",
+    "lint-staged": "^7.0.2",
+    "prettier": "^1.11.1",
     "react": "^16.2.0",
```
Comment (alextrastero) [inline package.json]: these should be `devDependencies`

File: .eslintrc [hunk 2]
```diff
@@ -3,13 +3,14 @@
 extends:
   - standard
   - standard-react
+  - prettier
```
Comment (alextrastero) [inline .eslintrc]: do we need `standard` if we have `prettier`? I don't know if some/all rules are redundant

Comment (onluiz) [reply .eslintrc]: I will remove standard and make some tests. But I think you're right.

File: package.json (hunk 1 above)
Comment (onluiz) [reply package.json]: Done =)

**Comment by onluiz** [comment]
@alextrastero, here is what I did:

- Adjusted prettier dependencies in package.json. Now they're into dev dependencies
- removed standard from eslitrc, keeping standard-react and prettier
- added eslint-plugin-prettier and eslint-config-prettier as devDependencies
- ran all tests a couple of times to make sure no new warning was shown

Could you review this? Want to make sure it is done the right way =)

Thank you very much!

**Comment by alextrastero** [comment]
Thank you! Going to have a look asap

**Comment by alextrastero** [comment]
`npm run lint` doesn't work, it's not using the rules that prettier uses

**Comment by alextrastero** [comment]
Also now the fact that it's a commit hook you can't see code diff in the console

**Comment by onluiz** [comment]
So, better remove commit book?

Also, let me fiz npm run lint.

**Comment by alextrastero** [comment]
If it's ok with you I would rather keep single quotes and double for JSX

**Comment by alextrastero** [comment]
So as far as I understand, now we have the choice to run `npm run prettier` and that would format our code

**Comment by alextrastero** [comment]
I don't want to be a bummer but I think I would rather make `npm test` fail if `npm run lint` fails (so remove the `; exit 0`)

**Comment by onluiz** [comment]
Haha that’s fine. Let me review it all and make another push.

We keep trying until it gets good. So please keep telling me what needs to be done. :)

**Comment by onluiz** [comment]
> If it's ok with you I would rather keep single quotes and double for JSX
Done

> remove the ; exit 0
Done

**Comment by alextrastero** [comment]
Perfect

**Pull request merged**